### ISSUE TITLE
Error with unresolved symlinks

### ISSUE THREAD

**Issue by pizzard** [RedRise/lm#11]
If there exists any symbolic symlinks which just point into nowhere (nonexisting directories or files), this causes the script to crash with the following error: 
File "/usr/local/bin/lm", line 1418, in <module>
    files = LM.get_files(args)
  File "/usr/local/bin/lm", line 1166, in get_files
    real_path, True, *self.file_ext ) )
  File "/usr/local/bin/lm", line 110, in filelist
    result.extend( filelist( f, True, *ext) )
  File "/usr/local/bin/lm", line 110, in filelist
    result.extend( filelist( f, True, *ext) )
  File "/usr/local/bin/lm", line 104, in filelist
    alist = [ os.path.abspath(os.path.join(dir,f)) for f in os.listdir( dir )]
OSError: [Errno 2] No such file or directory: /nonexisting/testdir

These links should be silently ignored.

**Comment by RedRise** [comment]
Thank you for this remark.
You're right, I need to improve directories parsing (and find the better option, between os.walk and os.listdir for example).
I'll keep you posted.

**Comment by RedRise** [comment]
Could you test the last version?

**Comment by pizzard** [comment]
I did, created random invalid symlinks to non-existing files, directories and so on and everything worked fine. Also hardlinks to devices not mounted etc.
But while testing this I saw another issue: If I create zero-byte files like test.mkv and apply lm it retrieves hash information an tells me that it has found an match an displays the movie The Recruit in orange.
This is definitely wrong for a 0-byte file. Wouldn't it be useful to skip zero byte file, because while my Browser or other programs when are still downloading they always create a 0-byte file of file name already.
